utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -823,6 +831,29 @@ impl<'a, 'b> SyscallsInvocator<'a, 'b> {
         }
     }
 
+    fn store_waited_message_id(&self, instructions: &mut Vec<Instruction>) {
+        let Some(gr_message_id_indexes_handle) = self
+            .syscalls_imports
+            .get(&InvocableSyscall::Loose(SyscallName::MessageId))
+            .map(|&(_, call_indexes_handle)| call_indexes_handle as u32)
+        else {
+            // We automatically enable the `message_id` syscall import if the `wait` syscall is enabled in the config.
+            // If not, then we don't need to store the message ID.
+            return;
+        };
+
+        let memory_layout = MemoryLayout::from(self.memory_size_bytes());
+        let start_offset = memory_layout.waited_message_id_ptr;
```

Comment:
won't that overwrite other resources?

@StackOverflowExcept1on ptal

Reply:
The macro for memory_layout is designed so that you can easily add new pointers, but this may change the output of wasm-gen.